From 17d0da9815466516a8daae8e6d5c6d797caa75fc Mon Sep 17 00:00:00 2001
From: Justin Waters <justin.waters@timesys.com>
Date: Thu, 10 Mar 2016 15:26:30 -0500
Subject: [PATCH 5/7] board: ge: bx50v3: Fix bootargs for ELO board

---
 include/configs/ge_bx50v3.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/configs/ge_bx50v3.h b/include/configs/ge_bx50v3.h
index 8be4350..5219746 100644
--- a/include/configs/ge_bx50v3.h
+++ b/include/configs/ge_bx50v3.h
@@ -35,9 +35,11 @@
 #if defined(CONFIG_TARGET_GE_ELO)
 #define CONFIG_MXC_UART_BASE	UART4_BASE
 #define CONFIG_CONSOLE_DEV	"ttymxc3"
+#define CONFIG_EXTRA_BOOTARGS	"panic=10"
 #else
 #define CONFIG_MXC_UART_BASE	UART3_BASE
 #define CONFIG_CONSOLE_DEV	"ttymxc2"
+#define CONFIG_EXTRA_BOOTARGS	"cma=128M"
 #endif
 
 #ifdef CONFIG_YOCTO_IMAGE
@@ -197,7 +199,7 @@
 			"echo 'U-Boot upgraded. Please reset'; " \
 		"fi\0" \
 	"setargs=setenv bootargs console=${console},${baudrate} " \
-		"root=/dev/${rootdev} rw rootwait cma=128M\0" \
+		"root=/dev/${rootdev} rw rootwait " CONFIG_EXTRA_BOOTARGS "\0" \
 	"loadbootscript=" \
 		"${loadcmd} ${dev} ${devnum}:${partnum} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from ${dev}:${devnum}:${partnum};" \
-- 
2.5.0

