From 660d9fe934bd3dcb0dde077d6f51adf8fc4fc779 Mon Sep 17 00:00:00 2001
From: Justin Waters <justin.waters@timesys.com>
Date: Tue, 19 Apr 2016 14:36:16 -0400
Subject: [PATCH 7/7] board: ge: bx50v3: Add correct SATA settings

Signed-off-by: Justin Waters <justin.waters@timesys.com>
---
 board/ge/bx50v3/bx50v3.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/board/ge/bx50v3/bx50v3.c b/board/ge/bx50v3/bx50v3.c
index 8d3e47a..18cac38 100644
--- a/board/ge/bx50v3/bx50v3.c
+++ b/board/ge/bx50v3/bx50v3.c
@@ -698,6 +698,30 @@ static iomux_v3_cfg_t const misc_pads[] = {
 #define SUS_S3_OUT	IMX_GPIO_NR(4, 11)
 #define WIFI_EN	IMX_GPIO_NR(6, 14)
 
+int setup_ba16_sata(void)
+{
+	struct iomuxc *const iomuxc_regs = (struct iomuxc *)IOMUXC_BASE_ADDR;
+	int ret;
+
+	ret = enable_sata_clock();
+	if (ret)
+		return ret;
+
+	clrsetbits_le32(&iomuxc_regs->gpr[13],
+			IOMUXC_GPR13_SATA_MASK,
+			IOMUXC_GPR13_SATA_PHY_8_RXEQ_3P0DB
+			|IOMUXC_GPR13_SATA_PHY_7_SATA2M
+			|IOMUXC_GPR13_SATA_SPEED_3G
+			|(1<<IOMUXC_GPR13_SATA_PHY_6_SHIFT)
+			|IOMUXC_GPR13_SATA_SATA_PHY_5_SS_DISABLED
+			|IOMUXC_GPR13_SATA_SATA_PHY_4_ATTEN_12_16
+			|IOMUXC_GPR13_SATA_PHY_3_TXBOOST_3P33_DB
+			|IOMUXC_GPR13_SATA_PHY_2_TX_1P133V
+			|IOMUXC_GPR13_SATA_PHY_1_SLOW);
+
+	return 0;
+}
+
 int board_early_init_f(void)
 {
 	imx_iomux_v3_setup_multiple_pads(misc_pads,
@@ -755,6 +779,10 @@ int board_late_init(void)
 	/* Backlight Power */
 	gpio_direction_output(LVDS_BACKLIGHT_GP, 1);
 
+#ifdef CONFIG_CMD_SATA
+	setup_ba16_sata();
+#endif
+
 	return 0;
 }
 
-- 
2.5.0

