From b555459597027af3362d93020162dfbb30e0eddd Mon Sep 17 00:00:00 2001
From: Ken Lin <ken.lin@advantech.com.tw>
Date: Sun, 13 Sep 2015 10:38:29 +0800
Subject: [PATCH 6/7] add usb hub_test suport

---
 drivers/usb/core/sysfs.c | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/drivers/usb/core/sysfs.c b/drivers/usb/core/sysfs.c
index d269738..e78188d 100644
--- a/drivers/usb/core/sysfs.c
+++ b/drivers/usb/core/sysfs.c
@@ -16,6 +16,8 @@
 #include <linux/usb/quirks.h>
 #include "usb.h"
 
+#include <uapi/linux/usb/ch11.h>
+
 /* Active configuration fields */
 #define usb_actconfig_show(field, format_string)			\
 static ssize_t field##_show(struct device *dev,				\
@@ -677,6 +679,30 @@ static ssize_t remove_store(struct device *dev, struct device_attribute *attr,
 static DEVICE_ATTR_IGNORE_LOCKDEP(remove, S_IWUSR, NULL, remove_store);
 
 
+
+static ssize_t usbhub_testmode_store(struct device *dev,
+               struct device_attribute *attr,
+               const char *buf, size_t count)
+{
+       int port, testmode;
+       struct usb_device *udev = to_usb_device(dev);
+
+       sscanf(buf, "%d %d\n", &port, &testmode);
+        printk(KERN_ERR "[Advantech]%s ,port:%d testmode:%d %d\n",  __func__, port, testmode, udev->portnum);
+
+       usb_control_msg(udev, usb_sndctrlpipe(udev, 0), USB_REQ_SET_FEATURE,
+                       USB_RT_PORT, USB_PORT_FEAT_SUSPEND,
+                       port, NULL, 0, 1000);
+       usb_control_msg(udev, usb_sndctrlpipe(udev, 0), USB_REQ_SET_FEATURE,
+                       USB_RT_PORT, USB_PORT_FEAT_TEST,
+                       (testmode << 8) | port, NULL, 0, 1000);
+	   
+			 return count;
+}
+
+static DEVICE_ATTR(hub_test, 0200, NULL, usbhub_testmode_store);
+
+
 static struct attribute *dev_attrs[] = {
 	/* current configuration's attributes */
 	&dev_attr_configuration.attr,
@@ -706,6 +732,9 @@ static struct attribute *dev_attrs[] = {
 	&dev_attr_remove.attr,
 	&dev_attr_removable.attr,
 	&dev_attr_ltm_capable.attr,
+        
+	&dev_attr_hub_test.attr,
+
 	NULL,
 };
 static struct attribute_group dev_attr_grp = {
-- 
2.8.1

